Add notes for 2022-{04-04, 05-16, 06-06, 08-15}, touch up existing notes (#291)
unicode-org/message-format-wg: meetings/2022/notes-2022-06-06.md

 
EAO: One extension point of this discussion is do we take the same approach for function names?  
 
MWS: I would like to clarify that the message formatting function should know what a `Person` type is.  It shouldn't know what types that the formatting function expects to handle.  The message library could generically handle that a Person object maps to a Map.  We don't need to require it, but we don't need to forbid it.
 
Regarding dots in a function name versus a variable name.  In a variable name, it is natural to allow the dots to signify a multi-level lookup.  For functions, one thing that Addison liked was that dots could signify a custom function name, where the dots signify a namespace in the style of Java package names / Maven artifact group ids.  Back to variables, we should still allow a flat level lookup.
 
MIH: I am fine to say that this topic is just an implementation detail, and I will go ahead and implement in the ICU4J prototype whatever I think makes the most sense for Java.
 

> EAO: It does sound like the function name is a separate discussion.  One thing I do like for custom functions is some signifier that indicates that a function is a custom function.  In ___, a dash indicates that a namespaced name, and in a similar way, maybe a "." could be used to indicate a custom function.

Reviewer: ```suggestion
EAO: It does sound like the function name is a separate discussion.  One thing I do like for custom functions is some signifier that indicates that a function is a custom function.  In Fluent, a leading dash indicates that a message is a term, and in a similar way, maybe a "." could be used to indicate a custom function.
```

 
EAO: Given this time crunch, we should identify who will submit PRs to improving the existing syntax spec according to the decisions that we are making?  Specifically, who other than STA and myself will submit PRs to change the EBNF structure?  That needs to be happening in parallel for us to be reaching these goals.
 
STA: I can definitely can work on the EBNF.  For the decision mentioned of separating local variables from selectors, this is a topic that is tightly coupled with other issues, such as whether and how we delimit selectors.  They are all regarding syntax visuals, and maybe we should handle them altogether this time around.  As long as we reach some sort of agreement, I can go ahead the EBNF.
 
EAO: I think STA and I can coordinate on that.
 
EAO: As a last sort of question, as a starting point for next week, I still think we should start in text mode.  I know that MIH and STA and MWS think we should start in code mode.  Is there anyone else besides me who thinks we should start in text mode?
 
ECH: I agree to start in code mode.
 
RGN: I have a weak preference to start in code mode. 
 
 

>  
>  
>  

Reviewer: ```suggestion
```

### the "." in variable names (`{$foo.bar :func}` or `{$foo :func opt=$bar.baz}` or even in selectors / local variables
 
#### Consensus : 
 
 
Next Meeting Agenda : 
- start in code or text  - https://github.com/unicode-org/message-format-wg/issues/256 - Look at meeting notes
- markup elements: same as placeholder, or not?
- the "." in variable names (`{$foo.bar :func}` or `{$foo :func opt=$bar.baz}` or even in selectors / local variables -  Already discussed we need to record consensus  
 

>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  
>  

Reviewer: ```suggestion
```
